WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -85,12 +86,36 @@ public Round(Round blameOf) : this(blameOf.RoundParameters)
 				var weightCredentialResponse = WeightCredentialIssuer.HandleRequest(zeroWeightCredentialRequests);
 
 				alice.SetDeadlineRelativeTo(ConnectionConfirmationTimeout);
+				foreach (var op in alice.Coins.Select(x => x.Outpoint))
+				{
+					if (RemoveAlicesNoLock(x => x.Coins.Select(x => x.Outpoint).Contains(op)) > 0)
+					{
+						Logger.LogInfo("Updated Alice registration.");
+					}
+				}
```

Comment:
Having the logic here and in the arena too is clunky. The problem is:

1. I must remove alices before I add a new one.
2. I must make sure the alice creted is definitely registrable.

Maybe the Arena should be responsible the Round's Alice lists?